### ISSUE TITLE
[question] FGS223 Double Switch: Key held down and Key released

### ISSUE THREAD

**Issue by Drynael** [zwave-js/zwavejs2mqtt#1059]
Hello,

I´m not sure if this is a Bug or just my misunderstanding.

I have this Fibaro FGS223 Double Switch 2 (https://devices.zwave-js.io/?jumpTo=0x010f:0x0203:0x1000:3.2)

The Parameters for this Device has 4 different Options:

S1 switch - Send Scenes: Key pressed 1 time
S1 switch - Send Scenes: Key pressed 2 times
S1 switch - Send Scenes: Key pressed 3 times
S1 switch - Send Scenes: Key held down and Key released

Regardless of what i do 1,2,3 or Hold MQTT Value from node14/37/1/currentValue changes from `"value": false` to `"value": true` and back.

am I missing somenthing?

I did a Rediscover of the whole node.

I´m running zwavejs2mqtt in a docker container on a rasp3:
`zwavejs2mqtt: 3.2.1
zwave-js: 7.2.2
home id: 4031*
 home hex:0xf0*`

**Comment by robertsLando** [comment]
Could you try with version 3.3.0 ?

**Comment by Drynael** [comment]
Same behavior on 3.3.0

Key held down -> True Realeasing -> back to False
![image](https://user-images.githubusercontent.com/15029013/114731968-70c08680-9d42-11eb-908b-f59481751218.png)

**Comment by robertsLando** [comment]
Please attach logs file of both zwavejs and zwavejs2mqtt 

/cc @AlCalzone

**Comment by AlCalzone** [comment]
You're looking at the wrong value (Binary Switch). The parameter with "Scenes" in the name configure what gets sent with the Central Scene CC.

**Comment by Drynael** [comment]
I´m fairly new to this whole zwaveJS2MQTT topic. I used the nativ zwave Integration from HA.


From The Logs: (If you need the whole one i would need to truncate it first it´s 170mb)
where is the `zwavejs` located?

S1 Key Held down for 3 Seconds (Lights switches on until i release the key)
```
2021-04-15 14:20:35.926 INFO ZWAVE: Node 14: value updated: 37-1-currentValue false => true
2021-04-15 14:20:35.937 INFO ZWAVE: Node 14: value added: 37-1-targetValue => undefined
2021-04-15 14:20:35.940 INFO ZWAVE: Node 14: value added 14-37-1-targetValue => undefined
2021-04-15 14:20:38.674 INFO ZWAVE: Node 14: value updated: 50-1-value-66049 0 => 26.5
2021-04-15 14:20:38.925 INFO ZWAVE: Node 14: value updated: 37-1-currentValue true => false
2021-04-15 14:20:39.686 INFO ZWAVE: Node 14: value updated: 50-1-value-66049 26.5 => 0
```
S2 Key Hold down for  3 Seconds (Lights switches on until i release the key)
```
2021-04-15 14:20:40.624 INFO ZWAVE: Node 14: value updated: 37-2-currentValue false => true
2021-04-15 14:20:40.630 INFO ZWAVE: Node 14: value added: 37-2-targetValue => undefined
2021-04-15 14:20:40.634 INFO ZWAVE: Node 14: value added 14-37-2-targetValue => undefined
2021-04-15 14:20:42.800 INFO ZWAVE: Node 14: value updated: 50-2-value-66049 0 => 26.6
2021-04-15 14:20:43.425 INFO ZWAVE: Node 14: value updated: 37-2-currentValue true => false
2021-04-15 14:20:43.809 INFO ZWAVE: Node 14: value updated: 50-2-value-66049 26.6 => 0
```

S1 Key pressed 3x (Lights switches on)
```
2021-04-15 14:20:46.494 INFO ZWAVE: Node 14: value updated: 37-1-currentValue false => true
```

S2 Key pressed 3x (Lights switches on)
```
2021-04-15 14:20:48.092 INFO ZWAVE: Node 14: value updated: 37-2-currentValue false => true
```

S1 + S2 Key pressed 1x (Lights switches off)
```
2021-04-15 14:20:48.762 INFO ZWAVE: Node 14: value updated: 37-2-currentValue true => false
2021-04-15 14:20:49.103 INFO ZWAVE: Node 14: value updated: 50-1-value-66049 0 => 26.5
2021-04-15 14:20:49.383 INFO ZWAVE: Node 14: value updated: 50-2-value-66049 0 => 0
2021-04-15 14:20:50.075 INFO ZWAVE: Node 14: value updated: 37-1-currentValue true => false
```

MQTT Topic `node14/91/0/scene/001 or 002` has no changes.

I did observe the Messages in `homeassistant/_EVENTS/ZWAVE_GATEWAY-zwavejs2mqtt-eg/node` but there is also no values pointing on scenes.

**Comment by AlCalzone** [comment]
Please post a whole logfile of `zwave-js` so I can figure out if the commands are even sent.
https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/bug_report?id=export-logs -> point 1.

**Comment by Drynael** [comment]
[zwavejs2mqtt-store.zip](https://github.com/zwave-js/zwavejs2mqtt/files/6320196/zwavejs2mqtt-store.zip)
Node 14

I tested the Switch at 17:47

**Comment by AlCalzone** [comment]
Looks like your test was happening while the driver was waiting for a callback from the stick that never came. Not sure if some of the dropped secure messages in the meantime contained the Central Scene messages. Can you try again?

**Comment by Drynael** [comment]
[zwavejs2mqtt-store (1).zip](https://github.com/zwave-js/zwavejs2mqtt/files/6320767/zwavejs2mqtt-store.1.zip)
Did stop the container removed the logfiles and restartet the container.
Tested around 19:45 in the logfiles

**Comment by AlCalzone** [comment]
There aren't any Central Scene commands sent. Maybe you need to add Node 1 to one of the additional association groups. Can you try that?

**Comment by Drynael** [comment]
I could try it but i don´t know how i have to do this :/

**Comment by diijkstra** [comment]
tl;dr - This switch does not need any associations of controller for scenes to work. Had the same issue on one of the switches, factory resetting helped.

Long story:
I've 3 such switches and had the very same issue. The scenes were being sent correctly on the one migrated with scenes from OZW, but did not work on the other one. Notifications where working, but scenes were not. After performing factory reset of the switch and re-configuring the switch to allow scenes (remember to check button type), finally the scenes were being send to the controller. It seems this might be an issue with your switch. 

I was able to check with PC Controller app, and the switch itself was not sending scene events to the controller (even when configuration values were correctly set by the zwavejs). I don't know if this is related, but it finally started working when I've used '15' as value to enable scenes for the single click on given button (just for fun, logs indicated that correct bits are always set, and the switch itself did report correct value - 0x0F in my case for enabling all the scenes).

**Comment by Drynael** [comment]
I did exclude -> Factory reset -> Include the module. 
The Scences still don´t work.

Latest Logs
The switch is now Node_39
[zwavejs2mqtt-store (2).zip](https://github.com/zwave-js/zwavejs2mqtt/files/6331130/zwavejs2mqtt-store.2.zip)

**Comment by AlCalzone** [comment]
@Drynael did you configure the device again after factory reset? The log doesn't contain everything after the inclusion, so I can't tell.

**Comment by Drynael** [comment]
I did a restart in between Inclusion and Test.

This is my current config:
![node_39](https://user-images.githubusercontent.com/15029013/115149903-6f56cd00-a066-11eb-9b14-a975457b0a62.png)
[node_39(JSON).zip](https://github.com/zwave-js/zwavejs2mqtt/files/6331504/node_39.JSON.zip)

**Comment by AlCalzone** [comment]
You didn't enable the scenes:
![grafik](https://user-images.githubusercontent.com/17641229/115150186-96fa6500-a067-11eb-8e10-3d919b059314.png)

**Comment by Drynael** [comment]
I have set them to 1. Which components do I have to reload/restart to take effect?

**Comment by AlCalzone** [comment]
None. These are set on the device and should have an effect immediately once they are accepted.

**Comment by Drynael** [comment]
I did some more testing. 
Press and Hold and Press 2x does nothing. But when i hit it Once i get a Scene Value of 3. But Value 3 should be 
```
    {
      "text": "KeyPressed2x",
      "value": 3
    },
```
![image](https://user-images.githubusercontent.com/15029013/115161376-a1cfec80-a09d-11eb-80a1-221496fa83bc.png)



Config:
![image](https://user-images.githubusercontent.com/15029013/115161283-1fdfc380-a09d-11eb-9393-936e457c7dba.png)

Logs:
[zwavejs2mqtt-store (4).zip](https://github.com/zwave-js/zwavejs2mqtt/files/6332225/zwavejs2mqtt-store.4.zip)

**Comment by diijkstra** [comment]
Take a look at the parameter 20 - Switch type. Only with 0 (monostable) all actions are send. With default value of 2, hold/release are not send. With value 1 neither hold/release nor the double click are being sent.

If you are getting double press on one click and no hold, this would be consistent with monostable button being interpreted as bistable - default value of parameter 20 being in use.

**Comment by Drynael** [comment]
I have now played around with alle the parameters.

Paramater 20: 0 - momentary switch 
Scenes are still configured as above
Central Scene slowRefresh can´t be switched off:
![image](https://user-images.githubusercontent.com/15029013/115278486-caabbc80-a145-11eb-944b-fc70afcd1348.png)
```yaml
2021-04-19 17:29:29.221 INFO ZWAVE: Writing false to 39-91-0-slowRefresh
2021-04-19 17:29:29.231 ERROR ZWAVE: Error while writing false on 39-91-0-slowRefresh: This virtual node does not support the command ConfigurationSet!
2021-04-19 17:29:29.234 ERROR ZWAVE: Unable to write false on 39-91-0-slowRefresh
2021-04-19 17:29:29.236 INFO ZWAVE: Success zwave api call writeValue { success: true, message: 'Success zwave api call', result: undefined }
```

If i press and Hold i get first Scene 2 (and lights turn on) and by releasing scene 1 (lights turn off) Should be only Scene ID 8
If i press 2x i get Scene 3 (Lights toggle) Should only be Scene ID 2

![image](https://user-images.githubusercontent.com/15029013/115278605-efa02f80-a145-11eb-8a64-fe2f2ef66380.png)

**Comment by AlCalzone** [comment]
> Central Scene slowRefresh can´t be switched off

That just means you're not getting 1000 "Held down" notifications while the button is held down. Also, I think this is just a display bug (@robertsLando), the logs show that slow refresh is not supported. So that value is most likely `undefined`.

> Paramater 20: 0 - momentary switch

Your log says this is set to `2` and it is never set differently (at least not in the logs):
```
2021-04-18T21:05:47.404Z CNTRLR « [Node 039] parameter #20 has value: 2
```

Did you actually perform the config changes (click on the arrow buttons)?

**Comment by AlCalzone** [comment]
Ok help me out here. What are you expecting to happen and what is actually happening? I'm seeing both Hold->Release and Pressed 2x notifications for scenes 1 and 2 in your latest log.

**Comment by Drynael** [comment]
I´m expecting that if i hit the Button once my Lights turn on and Off.
and if I hit 2x or Hold it sends only the scene so i can use them to turn on/off other devices.
What i get is Regardless of what combination of key Presses is use. The lights turn on and off or do nothing at all.

For Example i try to Turn on Ambient Lights with PressAndHold. I get now the input KeyHold (and release). I can use the Scene information to Toggle the Ambient Lights. What irritates me is that the ceiling lights turns on when I activate one of the scenes. Maybe I am making the wrong assumption that the module can send scenes without switching.

**Comment by diijkstra** [comment]
@Drynael Unfortunately the switch will always toogle the output when pressing the button (1, 2, 3 presses and the hold), this is a limitation of this hardware (see: https://forum.fibaro.com/topic/26631-fgs-223-turn-on-associated-device-by-double-click/) . There is nothing you can do about it on this particular device type.

You could workaround it by automating that the load will be toggled after the scene is received on ZwaveJS/Mqtt/HA side - but due to delays, the change will still be visible to the operator.

**Comment by AlCalzone** [comment]
Yup, that sounds like a hardware limitation.

**Comment by Drynael** [comment]
Thank you all for figuring out all.
I can't find any switch with the capability to just send a scene and not triggering the lights.

I just use some regular Scene managers or Remotes.

**Comment by diijkstra** [comment]
Same here, I've moved crucial switches to Shelly 2.5 - with HA the delay between synchronizing zwave devices is negligible and it works like a charm with Tasmota and MQTT.
